tests/__init__.py

```diff
@@ -1,40 +1,120 @@
-from typing import Any, Dict, Optional
+# Copyright 2022 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sqlite3
+from asyncio import Future
+from typing import Any, Awaitable, Callable, Optional, Tuple, TypeVar
+from unittest.mock import Mock
 
-import attr
-from mock import Mock
-from synapse.module_api import ModuleApi
+from synapse.module_api import JsonDict, ModuleApi
 
 from tchap_red_list import RedListManager
 
+RV = TypeVar("RV")
+TV = TypeVar("TV")
 
-@attr.s(auto_attribs=True)
-class MockEvent:
-    """Mocks an event. Only exposes properties the module uses."""
-    sender: str
-    type: str
-    content: Dict[str, Any]
-    room_id: str = "!someroom"
-    state_key: Optional[str] = None
 
-    def is_state(self) -> bool:
-        """Checks if the event is a state event by checking if it has a state key."""
-        return self.state_key is not None
+class SQLiteStore:
+    """In-memory SQLite store. We can't just use a run_db_interaction function that opens
+    its own connection, since we need to use the same connection for all queries in a
+    test.
+    """
+
+    def __init__(self) -> None:
+        self.conn = sqlite3.connect(":memory:")
+
+    async def run_db_interaction(
+        self, desc: str, f: Callable[..., RV], *args: Any, **kwargs: Any
+    ) -> RV:
+        cur = CursorWrapper(self.conn.cursor())
+        try:
+            res = f(cur, *args, **kwargs)
+            self.conn.commit()
+            return res
+        except Exception:
+            self.conn.rollback()
+            raise
+
+
+class MockEngine:
+    supports_using_any_list = False
+
+
+class CursorWrapper:
+    """Wrapper around a SQLite cursor."""
```

Comment:
What's the motivation for adding `MockEngine` and `CursorWrapper`?

Reply:
We need the cursor to have a `database_engine` property, which is an object which `supports_using_any_list` equals `False` (since we're using SQLite). This is because `DatabasePool.simple_select_many_txn` checks it to figure out how it should build the SQL query. This is part of the wrapping around cursors that is normally done by Synapse.